src/validators/function.rs

```diff
@@ -86,10 +86,14 @@ impl Validator for FunctionBeforeValidator {
         slots: &'data [CombinedValidator],
         recursion_guard: &'s mut RecursionGuard,
     ) -> ValResult<'data, PyObject> {
-        let kwargs = kwargs!(py, data: extra.data, config: self.config.clone_ref(py), context: extra.context);
+        let info = ValidatorInfo {
```

Comment:
can you add a `new` method to `ValidatorInfo`, then the signature can be `ValidatorInfo::new(extra, self.config.clone_ref(py))`.

Reply:
done